examples/demo-rollup/src/main.rs

```diff
@@ -139,23 +119,22 @@ async fn main() -> Result<(), anyhow::Error> {
         ledger_rpc::get_ledger_rpc::<DemoBatchReceipt, DemoTxReceipt>(ledger_db.clone());
     methods
         .merge(ledger_rpc_module)
-        .expect("Failed to merge rpc modules");
+        .expect("Failed to merge ledger RPC modules");
+
+    let batch_builder = demo_runner.take_batch_builder()?;
+
+    let r = get_txs_rpc::<
```

Comment:
I'm surprised we need type annotations here - generally, we want to avoid references to module-system specific structs like `DefaultContext` in the the demo. Do you know why type inference is broken here?